frontend/src/metabase/visualizations/types/visualization.ts

```diff
@@ -176,24 +193,59 @@ export type VisualizationSettingDefinition<TValue, TProps = void> = {
   getMarginBottom?: (series: Series, settings: VisualizationSettings) => string;
   persistDefault?: boolean;
   inline?: boolean;
-  props?: TProps;
-  getProps?: (
-    series: Series,
-    vizSettings: VisualizationSettings,
-    onChange: (value: TValue) => void,
-    extra: unknown,
-    onChangeSettings: (value: Record<string, any>) => void,
-  ) => TProps;
   readDependencies?: string[];
   writeDependencies?: string[];
   eraseDependencies?: string[];
-  // is the setting visible in the dashboard card viz settings
   dashboard?: boolean;
   useRawSeries?: boolean;
+  hidden?: boolean;
+  placeholder?: string;
+  index?: number;
 };
 
+export type WidgetComponentDef<Widget, P, TValue> = {
+  widget?: Widget;
+  props?: P;
+  getProps?: (
+    series: Series,
+    vizSettings: VisualizationSettings,
+    onChange: (value: TValue | null) => void,
+    extra: SettingDefinitionExtra,
+    onChangeSettings: (value: VisualizationSettings) => void,
+  ) => P;
+};
+
+export type WidgetStringDef<Widget, TValue> = {
+  widget?: Widget;
+  props?: unknown;
+  getProps?: (
+    series: Series,
+    vizSettings: VisualizationSettings,
+    onChange: (value: TValue | null) => void,
+    extra: SettingDefinitionExtra,
+    onChangeSettings: (value: VisualizationSettings) => void,
+  ) => unknown;
+};
```

Comment:
I wonder if we can piggy-back on `WidgetComponentDef` here, as the declaration is almost the same, just with `unknown` as Props. We don't have a generic type constraint on `WidgetComponentDef` so this should work.

```suggestion
export type WidgetStringDef<Widget, Value> = WidgetComponentDef<Widget, unknown, Value>
```